### PR TITLE
Module/pathseq/1.0

### PR DESCRIPTION
**Pull request by Kdreval**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [X] I used the cookiecutter template and updated the placeholder rules.

- [X] The snakemake rules follow the [design guidelines](https://lcr-modules.readthedocs.io/en/latest/for_developers.html#module-rules). 

  - [X] All references to the `rules` object (_e.g._ for input files) are wrapped with `str()`.

- [X] Every rule in the module is either listed under `localrules` or has the `threads` and `resources` directives.

- [X] Input and output files are being symlinked into the `CFG["inputs"]` and `CFG["outputs"]` subdirectories, respectively.

- [X] I updated the final target rule (`*_all`) to include every output rule.

- [X] I explained important module design decisions in `CHANGELOG.md`.

- [X] I updated the `default.yaml` configuration file to provide default values for each rule in the module snakefile.

- [X] I did not set any global wildcard constraints. Any/all wildcard constraints are set on a per-rule basis. 

- [X] I ensured that all symbolic links are relative and self-contained (_i.e._ do not point outside of the repository) by specifying `in_module=True` or using absolute symlink.

- [X] I replaced every value that should (or might need to) be updated in the default configuration file with `__UPDATE__`.

- [X] I recursively searched for all comments containing `TODO` to ensure none were left. For example:

  ```bash
  grep -r TODO modules/<module_name>/1.0
  ```

### If applicable

- [X] I added more granular output subdirectories.

- [ ] I added rules to the `reference_files` workflow to generate any new reference files.

- [ ] I added subdirectories with large intermediate files to the list of `scratch_subdirectories` in the `default.yaml` configuration file.

- [ ] I updated the list of available wildcards for the input files in the `default.yaml` configuration file.

## Checklist for Updated Module

To be completed.

### CODE REVIEW

File: modules/pathseq/1.0/pathseq.smk [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    print('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}' + '\x1b[0m')
+    print('\x1b[0;31;40m' + f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m')
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
```
Comment (oncogenomics) [inline modules/pathseq/1.0/pathseq.smk]: Replace this with Lakshay's up to date warning log:
```current_version = pkg_resources.get_distribution("oncopipe").version
if version.parse(current_version) < version.parse(min_oncopipe_version):
    logger.warning(
                '\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}'
                "\n" f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m'
                )
    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")```

**Review by oncogenomics — COMMENTED** [review]
Just a couple small changes to make, you can also run ```snakemake --lint``` if you haven't already.

**Comment by Kdreval** [comment]
I can confirm it is running after introducing changes suggested in the PR review and it is ready for re-review.

**Review by Lakshay-sethi — APPROVED** [review]
1. default.yaml
Examples are a very nice touch
2. changelog
Very informative, good job

**Comment by Kdreval** [comment]
@Lakshay-sethi  can you please update if you included this module in the demo and did the test runs?

File: modules/pathseq/1.0/pathseq.smk [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+    _pathseq_output,
+    _pathseq_all
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _pathseq_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"]
```
Comment (lkhilton) [inline modules/pathseq/1.0/pathseq.smk]: Could you please include the bai as an input to this rule, too? It ensures that any alignment modules that might be run upstream complete their index-producing steps.

**Review by lkhilton — CHANGES_REQUESTED** [review]
Looks great! Requesting one tiny change to the input bam file rule.

File: modules/pathseq/1.0/pathseq.smk (hunk 2 above)
Comment (Kdreval) [reply modules/pathseq/1.0/pathseq.smk]: Thanks! I did include the bai index as input and pushed the updates, so this is ready for re-review.

**Review by lkhilton — APPROVED** [review]

**Pull request merged**